src/Workspaces/Core/Portable/Shared/Utilities/AsyncBatchingWorkQueue`2.cs

```diff
@@ -140,7 +141,15 @@ void AddItemsToBatch(IEnumerable<TItem> items)
             async Task<TResult?> ContinueAfterDelay(Task lastTask)
             {
                 using var _ = _asyncListener.BeginAsyncOperation(nameof(AddWork));
-                await lastTask.ConfigureAwait(false);
+                try
+                {
+                    await lastTask.ConfigureAwait(false);
+                }
+                // Make sure the task being performed doesn't propagate out a cancellation exception that isn't
+                // associated with the token we pass into it.
```

Comment:
Would prefer this comment be inside the `catch` block, with a comment that we don't want an exception thrown by a work item to break the queue.